woocommerce/woocommerce
# Core profiler: set default for props to avoid crashes

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This fix sets default for the properties of the onboarding profile from the context, solving an issue where if some of them are undefined, it will prevent the user from proceeding with the core profiler.

Closes #49512, which has a description of the issue.

Implementing this fix allowed me to have this screen again, where I was able to skip this step:

<img width="590" alt="Bildschirmfoto 2024-07-13 um 22 18 14" src="https://github.com/user-attachments/assets/fba92208-8687-4d97-b766-4ef405fb9cfe">

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a fresh site using this branch with Code Snippets plugin installed
2. Add the following code snippets:

```php
add_filter( 'woocommerce_rest_prepare_data_country', function() {
    throw new Exception('Error');
});

add_filter( 'woocommerce_rest_api_option_permissions', function() {
    return new Exception('Error');
});
```

2. Go to core profiler
3. Continue to `Tell us a bit about your store` step
4. Check that `Oops! We encountered a problem while fetching the list of countries to choose from. Please try again or skip this step` error message is displayed
5. Click on `Please try again` button
6. Confirm that the page is reloaded and the error message is displayed again without crashing the site
7. Click on `Skip this step` button and confirm that the profiler continues to the next step

### Changelog entry


<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message Failure to fetch the country list no longer blocks users in the profiler.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
